spring-integration-ip/src/test/java/org/springframework/integration/ip/tcp/connection/TcpNioConnectionReadTests.java

```diff
@@ -411,8 +474,15 @@ private void testClosureMidMessageGuts(AbstractByteArraySerializer serializer, S
 		final Semaphore semaphore = new Semaphore(0);
 		final List<TcpConnection> added = new ArrayList<TcpConnection>();
 		final List<TcpConnection> removed = new ArrayList<TcpConnection>();
+
+		final CountDownLatch errorMessageLetch = new CountDownLatch(1);
+		final AtomicReference<Throwable> errorMessageRef = new AtomicReference<Throwable>();
+
 		AbstractServerConnectionFactory scf = getConnectionFactory(serializer, message -> {
-			responses.add(message);
```

Comment:
Looks like we should have removed this variable - noticed during conflict resolution during cherry-pick.